File: Euclid/Game/Catalogue/CatalogueManager.cs
using Euclid.Storage.Database.Access;
using Euclid.Storage.Database.Data;
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.Linq;

namespace Euclid.Game
{
    public class CatalogueManager : ILoadable
    {
        #region Fields

        public static readonly CatalogueManager Instance = new CatalogueManager();

        #endregion

        #region Properties

        public List<CatalogueItem> Offers { get; private set; }
        public ConcurrentDictionary<int, CatalogueOrder> OrderHistory { get; private set; }

        #endregion

        #region Constructors

        public void Load()
        {
            Offers = CatalogueDao.GetCatalogueOffers().Select(x => new CatalogueItem(x)).ToList();
            OrderHistory = new ConcurrentDictionary<int, CatalogueOrder>();
        }

        #endregion

        #region Public methods


        #endregion
    }
}

File: euclid/src/main/java/euclid/game/catalogue/CatalogueManager.java
package euclid.game.catalogue;

import euclid.game.ILoadable;
import euclid.storage.database.access.CatalogueDao;
import euclid.storage.database.data.CatalogueData;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.ConcurrentHashMap;

public class CatalogueManager implements ILoadable {

    public static final CatalogueManager INSTANCE = new CatalogueManager();

    private List<CatalogueItem> offers;
    private ConcurrentHashMap<Integer, CatalogueOrder> orderHistory;

    private CatalogueManager() {
    }

    public static CatalogueManager getInstance() {
        return INSTANCE;
    }

    @Override
    public void load() {
        offers = new ArrayList<>();
        List<CatalogueData> catalogueOffers = CatalogueDao.getCatalogueOffers();
        for (CatalogueData data : catalogueOffers) {
            offers.add(new CatalogueItem(data));
        }
        orderHistory = new ConcurrentHashMap<>();
    }

    public List<CatalogueItem> getOffers() {
        return offers;
    }

    public ConcurrentHashMap<Integer, CatalogueOrder> getOrderHistory() {
        return orderHistory;
    }
}
